Adding new function, narrative method, and re-naming UploaderUtil

samseaver opened the pull request.

samseaver commented:
OK, I fixed the conflicts, can you re-merge and run your tests to be sure?

Tianhao-Gu commented on lib/kb_uploadmethods/UploaderUtil.py:
```diff
@@ -0,0 +1,262 @@
+import os
```
This file should be deleted.

Tianhao-Gu commented on lib/kb_uploadmethods/kb_uploadmethodsImpl.py:
```diff
@@ -113,7 +113,7 @@ def upload_fastq_file(self, ctx, params):
             returnVal = fastqUploader.upload_fastq_file(params)
 
-        reportVal = fastqUploader.generate_report(returnVal['obj_ref'], params)
+        reportVal = Uploader.generate_report(returnVal['obj_ref'], params)
```
it should still be 'fastqUploader'

Tianhao-Gu commented on lib/kb_uploadmethods/kb_uploadmethodsImpl.py:
```diff
@@ -124,6 +124,40 @@ def upload_fastq_file(self, ctx, params):
+
+        #Text entry used, stripping whitespace
+        for key, value in params.iteritems():
+            params[key] = value.strip()
```
if user accidentally passed a number value, this will break.

Tianhao-Gu commented on ui/narrative/methods/load_genome_from_fasta_gff/display.yaml:
```diff
@@ -0,0 +1,41 @@
+            []
+
+parameters :
+    fasta_file
```
needs ':'

Tianhao-Gu commented:
i put an initial review. You can fix those while you are finishing definition for the new methods.

samseaver commented:
Just leaving a note here to say I addressed these comments.

Tianhao-Gu commented:
I didn't see any changes.

samseaver commented:
I haven't committed the changes, because there's also commits for the actual functionality itself. I left the note to say that amongst those commits, I addressed your comments. I'll commit the changes when I get it working with the latest kb_gffupload repository tomorrow.

Tianhao-Gu commented on lib/kb_uploadmethods/Utils/UploaderUtil.py:
```diff
@@ -52,6 +53,25 @@ def upload_fastq_file(self, params):
+      genome_name: output genome object name
+      workspace_name: workspace name that genome will be stored to
+      """
+      log('--->\nrunning UploaderUtil.upload_gff_fasta_file\nparams:\n%s' % json.dumps(params, indent=1))
```
breaking pep8. max line length is 92.

Tianhao-Gu commented on lib/kb_uploadmethods/Utils/UploaderUtil.py:
```diff
@@ -52,6 +53,25 @@ def upload_fastq_file(self, params):
+      """
+      log('--->\nrunning UploaderUtil.upload_gff_fasta_file\nparams:\n%s' % json.dumps(params, indent=1))
+      
+      #Test file availability
```
remove those comments.

Tianhao-Gu commented on lib/kb_uploadmethods/Utils/UploaderUtil.py:
```diff
@@ -52,6 +53,25 @@ def upload_fastq_file(self, params):
 
     return returnVal
 
+  def upload_gff_fasta_file(self, params):
```
Should do a param validation check here.

Tianhao-Gu commented on lib/kb_uploadmethods/Utils/UploaderUtil.py:
```diff
@@ -251,6 +271,29 @@ def _upload_file_path(self, params):
+      
+      #for now must use workspace name, but no ws_id_to_name() function available
+      if str(workspace_name_or_id).isdigit():
+          raise ValueError('"' + workspace_name_or_id + '" parameter is a workspace id and workspace name is required')
```
pep8

Tianhao-Gu replied:
I don't see a test for this.

samseaver replied:
Do you mean a unittest or in parameter validation?

Tianhao-Gu replied:
not unittest, in your actual methods. 

Then in unittest, test to see if bad params triggers the error.

Tianhao-Gu commented on test/gff_fasta_plant_upload_test.py:
```diff
@@ -0,0 +1,107 @@
+        shutil.copytree(data_dir, scratch_data_dir)
+
+        fasta_file = "Test_v1.0.fa.gz"
+        gff_file = "Test_v1.0.gene.gff3.gz"
```
remove unused test files

samseaver replied:
Are you talking about the other files? They'y'll be integrated at some point as we expand on what a user can do with a genome object.

Tianhao-Gu commented on test/gff_fasta_plant_upload_test.py:
```diff
@@ -0,0 +1,107 @@
+                'workspace_name':ws_name,
+                'genome_name':ws_obj_name
+            })[0]
+        pprint(result)
```
print is not a valid test

Tianhao-Gu commented on ui/narrative/methods/load_genome_from_fasta_gff/spec.json:
```diff
@@ -0,0 +1,95 @@
+          "target_property": "wsName"
+        },
+        {
+          "service_method_output_path": [0,"obj_ref"],
```
I don't think your func returns obj_ref, report_name nor report_ref
